hooks/pre/35-change-evr-extension-ownership.rb

```diff
@@ -0,0 +1,21 @@
+# In Katello 4.16, the 'evr' extension is removed from PostgreSQL and integrated into the Katello database via a migration.
+# This hook ensures the 'evr' extension's ownership is transferred to the 'foreman' user so migrations can act on it.
+
+if (local_foreman_db? || devel_scenario?) && execute("rpm -q postgresql-evr", false, false)
+  if app_value(:noop)
+    logger.debug("Would start postgresql service") if app_value(:noop)
+  else
+    start_services(['postgresql'])
+  end
+
+  # Update the ownership of the evr extension
+  database = param_value('foreman', 'db_database') || 'foreman'
+  username = param_value('foreman', 'db_username') || 'foreman'
+  sql = "psql -d '#{database}' -c \\\"UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{username}') WHERE extname='evr';\\\""
+  if app_value(:noop)
+    logger.debug("Would execute: #{sql} to update ownership of the evr extension")
+  else
+    logger.debug("Updating ownership of the evr extension")
+    execute_as!('postgres', sql, false, true)
```

Comment:
This has me thinking: how could this fail and how would a user resolve it?

One I can think of is that the user has `postgresql-evr` installed but the DB doesn't actually exist. I'm wondering if we need to catch that. Would that affect backup restores?

Any others that you can think of?

Reply:
Tested --reset-data and it was failing..I added a check to confirm if local foreman database exists before proceeding.

For backup-restore, I'd assume the supported workflow would be databases at the same version of foreman where the migrations in [katello PR](https://github.com/Katello/katello/pull/11159) have run.